File: WePeiYang/app/src/main/java/com/twt/service/home/tools/ToolsFragment.java
package com.twt.service.home.tools;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.GridLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.twt.service.R;
import com.twt.service.base.BaseFragment;
import com.twt.service.databinding.FragmentToolsBinding;

/**
 * Created by retrox on 2016/12/12.
 */

public class ToolsFragment extends BaseFragment {

    private FragmentToolsBinding binding;

    public static ToolsFragment newInstance() {

        Bundle args = new Bundle();

        ToolsFragment fragment = new ToolsFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_tools, container, false);
        binding.setViewModel(new ToolsFragViewModel(getContext()));
        return binding.getRoot();
    }

    @Override
    public void onLazyInitView(@Nullable Bundle savedInstanceState) {
        super.onLazyInitView(savedInstanceState);
        binding.recyclerView.setLayoutManager(new GridLayoutManager(getContext(), 4));

    }
}

File: WePeiYang/app/src/main/java/com/twt/service/home/tools/ToolsFragment.kt
package com.twt.service.home.tools

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.GridLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import com.twt.service.R
import com.twt.service.schedule2.view.exam.ExamTableActivity
import com.yookiely.lostfond2.waterfall.WaterfallActivity
import xyz.rickygao.gpa2.view.GpaActivity

/**
 * Created by retrox on 2016/12/12.
 */

class ToolsFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
            inflater.inflate(R.layout.fragment_tools, container, false).apply {
                findViewById<ImageView>(R.id.iv_fragment_tools_back).setOnClickListener { activity?.onBackPressed() }
                findViewById<RecyclerView>(R.id.recyclerView).apply {
                    layoutManager = GridLayoutManager(context, 4)
                    adapter = ToolsAdapter(context, listOf(
                            ToolItem(R.drawable.ic_main_schedule, "课程表", com.twt.service.schedule2.view.schedule.ScheduleActivity::class.java),
                            ToolItem(R.drawable.ic_main_gpa, "GPA", GpaActivity::class.java),
                            ToolItem(R.drawable.ic_main_lostfound, "失物招领", WaterfallActivity::class.java)
//                            ToolItem(R.drawable.ic_main_read, "考表", ExamTableActivity::class.java)
//                            ToolItem(R.drawable.ic_main_party, "党建", PartyActivity::class.java)
//                            ToolItem(R.drawable.ic_main_network, "上网", TjuNetActivity::class.java)
                    ))
                }
            }

}
